src/components/inputs/UserServiceUnitSelectInput.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import React, {Component} from 'react';
+import flowRight from 'lodash/flowRight';
+import {connect} from 'react-redux';
+import {reduxForm} from 'redux-form';
+import Select from 'react-select';
+
+import FormFieldLabel from '$components/form/FormFieldLabel';
+import DropdownIndicator from '$components/inputs/DropdownIndicator';
+import LoadingIndicator from '$components/inputs/SelectLoadingIndicator';
+import {FormNames} from '$src/enums';
+import {setUserActiveServiceUnit} from '$src/usersPermissions/actions';
+
+import type {UserServiceUnit, UserServiceUnits} from '$src/usersPermissions/types';
+
+type Props = {
+  userServiceUnits: UserServiceUnits,
+  userActiveServiceUnit: UserServiceUnit,
+  setUserActiveServiceUnit: Function,
+}
+
+class UserServiceUnitSelectInput extends Component<Props, State> {
+  handleChange = (val: any) => {
+    const {
+      setUserActiveServiceUnit,
+      userServiceUnits,
+    } = this.props;
+
+    if (val) {
+      const {value} = val;
+
+      const selected = userServiceUnits.filter((u) => u.id === value)[0];
```

Comment:
[Find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) function returns the first item found so instead of filtering and getting the first item by index, you can just use `find` to get the first found item.

Reply:
It returns the first item, but it is wrapped in an array, so that index access is needed.